### ISSUE TITLE
RPM packaging for F# doesn't include targets files in correct locations

### ISSUE THREAD

**Issue by dsyme** [mono/linux-packaging-fsharp#1]
A complete packaging of F# for Linux needs to have targets files in specific locations, because these locations are hardwired into the standard form of F# project files that need to cross-compile on multiple systems. 

The approach we use is simply to put soft links for the targets files Microsoft.FSharp.Targets and Microsoft.Portable.FSharp.Targets, and also FSharp.Build.dll in all of these locations:

```
Microsoft\ SDKs/F#/3.0/Framework/v4.0/...
Microsoft\ SDKs/F#/3.1/Framework/v4.0/...
xbuild/Microsoft/VisualStudio/v/FSharp/...
xbuild/Microsoft/VisualStudio/v12.0/FSharp/...
xbuild/Microsoft/VisualStudio/v14.0/FSharp/...
```

under the mono installation - these are soft links to the canonical copy.  Currently the RPM packaging only has one canonical copy of the targets file:

```
https://github.com/mono/linux-packaging-fsharp/blame/centos/fsharp.spec#L86
```

We use the full set of locations where targets files need to live. The paths originate from a range of tools (Visual Studio 2010, 2012, 2013) which have historically moved the location the targets file from version to version.

The lack of these files in the RPM package will mean that standard F# projects won't build, and project files will need to be adjusted.

The same applies to packaging for any other Linux target.

Thanks
Don

**Comment by rneatherway** [comment]
Hey, I just opened #2 for this.

Are you sure we need FSharp.Build.dll all over the place as well? I am not on Debian at the moment, but OSX works OK without it.

This script https://github.com/mono/linux-packaging-fsharp/blob/centos/fix-mono-4.0-installation.patch symlinks the genuine copies from 4.5 into 4.0, which is not right, but should still work.

Building [FsAutoComplete](https://github.com/fsharp/FsAutoComplete/blob/master/FsAutoComplete/FsAutoComplete.fsproj) should fail as it stands, but should work after my patch is applied. Note that I do not think my PR is the full complete fix, because I don't quite understand the interplay between the patches and the spec file. Do you clone upstream, apply the patches and then build the package? What is the exact sequence of commands?

**Comment by dsyme** [comment]
Closing this since it's a very old issue
